python/metatensor-core/metatensor/core/tensor.py

```diff
@@ -48,7 +48,16 @@ def __init__(self, keys: Labels, blocks: Sequence[TensorBlock]):
         :param keys: keys associated with each block
         :param blocks: set of blocks containing the actual data
         """
-        assert isinstance(keys, Labels)
+        if not isinstance(keys, Labels):
+            raise TypeError(f"`keys` must be metatensor Labels, not {type(keys)}")
```

Comment:
same here, might be better readable
```suggestion
            raise TypeError(f"`keys` must be metatensor Labels, not {type(keys)!r}")
```